src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncEHTests.cs

```diff
@@ -2104,5 +2104,148 @@ static void Main()
 ";
             CompileAndVerify(source, expected);
         }
+
+        [Fact, WorkItem(67091, "https://github.com/dotnet/roslyn/issues/67091")]
+        public void NestedCatch_DuplicateLocal()
+        {
+            var source = """
+                using System;
+                using System.Threading.Tasks;
+                using static System.Console;
+
+                class C
+                {
+                    static async Task Main()
+                    {
+                        await new C().M1();
+                    }
+
+                    int _counter;
+
+                    bool F(Exception ex)
+                    {
+                        _counter++;
+                        WriteLine($"F: {_counter} {ex.Message}");
+                        return _counter == 2;
+                    }
+
+                    async Task M1()
+                    {
+                        try
+                        {
+                            throw new Exception("M1");
+                        }
+                        catch (Exception ex) when (F(ex))
+                        {
+                            await M2(ex);
```

Comment:
>await M2(ex); [](http://example.com/codeflow?start=28&length=13)

I think it would be good to test invocation of `M1` that goes though this code path.  Perhaps result of `F` could be configurable and we invoke `M1` for two configurations.